
syntax = "proto3";

import "nanopb.proto";

message CoverConfigMessage {
	uint32 position = 1 [(nanopb).int_size = IS_8];
	uint32 direction = 2 [(nanopb).int_size = IS_8];
	uint32 hand_pull = 3 [(nanopb).int_size = IS_8];
	uint32 weak_switch = 4 [(nanopb).int_size = IS_8];
	uint32 power_switch = 5 [(nanopb).int_size = IS_8];
	uint32 report_interval = 6 [(nanopb).int_size = IS_16];
    uint32 led = 7 [(nanopb).int_size = IS_8];

    bool autoStroke = 9;
    
	uint32 pin_rx = 20 [(nanopb).int_size = IS_8];
	uint32 pin_tx = 21 [(nanopb).int_size = IS_8];
	uint32 pin_led = 22 [(nanopb).int_size = IS_8];
	uint32 pin_btn = 23 [(nanopb).int_size = IS_8];
}